src/lib/deploy.js

```diff
@@ -123,17 +137,107 @@ async function deploy({ network, yes }) {
     );
   }
 
-  await step('Generate verification key', async () => {
-    // TODO: BLOCKED Generate the verification key.
+  // Find the users file to import the smart contract from
+  let smartContractFile = await findSmartContractToDeploy(
+    `${DIR}/build/**/*.js`,
+    contractName
+  );
+
+  let smartContractImports;
+  try {
+    smartContractImports = await import(
+      `${DIR}/build/src/${smartContractFile}`
+    );
+  } catch (_) {
+    log(
+      red(
+        `  Failed to find the "${contractName}" smart contract in your build directory.\n Please confirm that your config.json contains the name of the smart contract that you desire to deploy to this network alias.`
+      )
+    );
+    return;
+  }
+
+  // Attempt to import the smart contract class to deploy from the users file. If we cannot find the named export
+  // log an error message and return early.
+  if (!(contractName in smartContractImports)) {
+    log(
+      red(
+        `  Failed to find the "${contractName}" smart contract in your build directory.\n Check that you have exported your smart contract class using a named export and try again.`
+      )
+    );
+    return;
+  }
+
+  // Attempt to import the private key from the `keys` directory. This private key will be used to deploy the zkapp.
+  let privateKey;
+  try {
+    privateKey = fs.readJSONSync(`${DIR}/keys/${network}.json`).privateKey;
+  } catch (_) {
+    log(
+      red(
+        `  Failed to find the the zkapp private key.\n  Please make sure your config.json has the correct 'keyPath' property.`
+      )
+    );
+    return;
+  }
+
+  let zkApp = smartContractImports[contractName]; //  The specified zkApp class to deploy
+  let zkAppPrivateKey = PrivateKey.fromBase58(privateKey); //  The private key of the zkApp
+  let zkAppAddress = zkAppPrivateKey.toPublicKey(); //  The public key of the zkApp
+
+  let verificationKey = await step('Generate verification key', async () => {
+    let { verificationKey } = await compile(zkApp, zkAppAddress);
+    return verificationKey;
   });
 
-  await step('Build transaction', async () => {
-    // TODO: BLOCKED Build the zkApp tx.
+  let partiesJsonDeploy = await step('Build transaction', async () => {
+    return JSON.parse(
+      await snarkyDeploy(zkApp, {
+        zkappKey: zkAppPrivateKey,
```

Comment:
@mitschabaude Could we make the deploy method expect `zkAppPrivateKey` please? For clarity of private key vs pub key.

Reply:
or `privateKey` since inside the deploy method, it's probably clear what that refers to.

Reply:
I think we should use "key" and "address" instead of privateKey and publicKey, because it's shorter and is what crypto people use. Also, the type of the parameter is PrivateKey, so it's impossible to accidentally pass in a public key. 

I also like to keep the "zkapp", because there is also the fee payer key involved in deploying, which is an optional parameter. Also, accidentally passing in the wrong private key is nothing the type system or runtime can protect against, so it's much more of a gotcha

Reply:
@mitschabaude publicKey and address are distinct and different. My main thing is `key` is a vague name as to whether it's the pub or private key. Also would like if it could follow `zkApp...` casing convention. But we can discuss more later. Doesn't need to block this PR.